kuma/tests/test_tag_injection.py

```diff
@@ -0,0 +1,155 @@
+    ],
+)
+def test_code_class_injection_edge_cases(aggregator, edge_code, setup_check):
+    """Test cases where code tags should not get code_class tags"""
+    metric_name = 'kuma.api_server.http_requests_inflight'
+    aggregator.assert_metric(metric_name)
+
+    # These should NOT have code_class - filter metrics with the specific code first
+    matching_metrics = [
+        metric_point for metric_point in aggregator.metrics(metric_name) if f'code:{edge_code}' in metric_point.tags
+    ]
+
+    assert matching_metrics, f"No metric found for {metric_name} with tag code:{edge_code}"
+
+    for metric_point in matching_metrics:
+        assert not any(
+            'code_class:' in tag for tag in metric_point.tags
+        ), f"Code:{edge_code} should not have code_class, got tags: {metric_point.tags}"
+
+
+def test_code_class_injection_no_code_label(aggregator, setup_check):
+    """Test that metrics without code label do not get code_class tags"""
+    metric_name = 'kuma.api_server.http_requests_inflight'
+    aggregator.assert_metric(metric_name)
+
+    found_metric = False
+    # No code label should not have code_class
+    for metric_point in aggregator.metrics(metric_name):
+        if 'handler:/no-code' in metric_point.tags:
+            found_metric = True
+            assert not any(
+                'code_class:' in tag for tag in metric_point.tags
+            ), f"Metric with handler:/no-code should not have code_class, got tags: {metric_point.tags}"
+    assert found_metric, f"No metric found for {metric_name} with tag handler:/no-code"
+
+
+@pytest.mark.parametrize(
+    'code,expected_class',
+    [
+        pytest.param('200', '2XX', id='2xx-ok'),
+        pytest.param('302', '3XX', id='3xx-found'),
+        pytest.param('401', '4XX', id='4xx-unauthorized'),
+        pytest.param('500', '5XX', id='5xx-internal'),
+        pytest.param('20', None, id='edge-2digit'),
+        pytest.param('2000', None, id='edge-4digit'),
+        pytest.param('abc', None, id='edge-non-numeric'),
+        pytest.param('2a0', None, id='edge-mixed'),
+        pytest.param('', None, id='edge-empty'),
+        pytest.param(None, None, id='edge-no-code'),
+    ],
+)
+def test_code_class_injection_unit(code, expected_class):
+    """Unit test for the _inject_code_class method"""
+
+    metric = CounterMetricFamily('test_metric', 'Test metric')
+    if code is not None:
+        sample = Sample('test_metric', {'code': code}, 1.0, None)
+    else:
+        sample = Sample('test_metric', {}, 1.0, None)
+    metric.samples = [sample]
+
+    modified_metric = KumaOpenMetricsScraper.inject_code_class(metric)
+    sample = modified_metric.samples[0]
+
+    if expected_class:
+        assert (
+            sample.labels.get('code_class') == expected_class
+        ), f"Expected code_class {expected_class} for code {code}"
+    else:
+        assert 'code_class' not in sample.labels, f"Expected no code_class for code {code}"
```

Comment:
Just to simplify the tests and avoid conditionals, the `get` method will return None already, we could probably just do 

```suggestion
    assert (
        sample.labels.get('code_class') == expected_class, 
        f"Expected code_class {expected_class} for code {code}"
    )
```

For the cases where code_class is None we would expect `code_class` not to be in the labels and therefore the get will return None.